src/UseCases/HandleCreditCardPaymentNotification/HandleCreditCardPaymentNotificationUseCase.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\UseCases\HandleCreditCardPaymentNotification;
+
+use Psr\Log\LoggerInterface;
+use WMDE\Fundraising\Frontend\Domain\Model\Donation;
+use WMDE\Fundraising\Frontend\Domain\Model\PaymentType;
+use WMDE\Fundraising\Frontend\Domain\Repositories\DonationRepository;
+use WMDE\Fundraising\Frontend\Domain\Repositories\GetDonationException;
+use WMDE\Fundraising\Frontend\Infrastructure\DonationAuthorizer;
+use WMDE\Fundraising\Frontend\Infrastructure\DonationConfirmationMailer;
+use WMDE\Fundraising\Frontend\Infrastructure\DonationEventLogger;
+
+/**
+ * @license GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class HandleCreditCardPaymentNotificationUseCase {
```

Comment:
How about `CreditCardNotificationUseCase`? For the NS I'd leave `Payment` in, though am dubious about the `Handle` part. That's what all UCs do right, handle a request :)